cli/push_test_event.php

```diff
@@ -0,0 +1,101 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Send events to event testing page
+ *
+ * @package    tool_realtime
+ * @subpackage cli
+ * @copyright  2020 Daniel Conquit, Matthew Gray, Nicholas Parker, Dan Thistlewaite
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+define('CLI_SCRIPT', true);
+
+require(__DIR__ . '/../../../../config.php');
+require_once("$CFG->libdir/clilib.php");
+require_once("$CFG->libdir/outputlib.php");
+
+$longparams = [
+    'context' => null,
+    'component' => null,
+    'area' => null,
+    'id' => null,
+    'payload' => null,
+    'help' => false
+];
+
+$shortmappings = [
+    'c' => 'context',
+    'n' => 'component',
+    'a' => 'area',
+    'i' => 'id',
+    'p' => 'payload',
+    'h' => 'help'
+];
+
+// Get CLI params.
+list($options, $unrecognized) = cli_get_params($longparams, $shortmappings);
+
+if ($unrecognized) {
+    $unrecognized = implode("\n  ", $unrecognized);
+    cli_error(get_string('cliunknowoption', 'admin', $unrecognized));
+}
+
+if ($options['help']) {
+    echo
+    "This CLI creates a new event with specific context and allows testing of the realtime tool
+Options:
+-c, --context       type of event to be sent
+-n, --component     identification number for event
+-a, --area          area related to event
+-i, --id            event ID
+-p, --payload       payload array for event
+-h, --help          Print out this help
+Example:
+
+\$ php admin/tool/realtime/cli/push_test_event.php --context=3\
+        --component=testcomponent --area=testarea --id=123 --payload='{\"testkey1\":\"testvalue1\",\"testkey2\":\"testvalue2\"}'\n";
+    die;
+}
+
+cli_heading('Push Test Event');
+
+if (!is_null($options['context'])) {
```

Comment:
I think for a prod ready cli script, we should probably make these 4 params either required or have sane defaults.

The context could default to context_system::instance()

and 'component' could default to 'moodle' (ie the core component') but the other things are really mandatory and if missing should show a cli_error()